### PR TITLE
hyper-v : Gen2 test automation fixes

### PR DESCRIPTION
**Pull request by asofron**

### CODE REVIEW

File: Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1 [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    )
+
+    $remoteScript = "BVT-NET-IFUP-IFDOWN.sh"
+
+    Set-Location $RootDir
+    #
+    # Run the guest VM side script to verify floppy disk operations
+    #
+    $stateFile = "${LogDir}\state.txt"
+    $BVT = "echo '${VMPassword}' | sudo -S -s eval `"export HOME=``pwd``;bash ${remoteScript} > BVT-NET-IFUP-IFDOWN.log`""
```
Comment (adriansuhov) [inline Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1]: $bvtCmd or $remoteCmd here. Variable name in this case should be lowercase and also we need to understand what it contains (a cmd to send on the VM)

File: Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1 [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        -downloadTo $LogDir -port $VMPort -username $VMUserName -password $VMPassword
+    $contents = Get-Content -Path $stateFile
+    if (($contents -eq "TestAborted") -or ($contents -eq "TestFailed")) {
+        LogErr "Error: Running $remoteScript script failed on VM!"
+        return "FAIL"
+    }
+    else {
+        LogMsg "Test BVT-NET-IFUP-IFDOWN PASSED !"
+    }
+
```
Comment (adriansuhov) [inline Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1]: unnecessary new line

File: Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1 [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    Start-Sleep 30
+    $newIP = Get-IPv4AndWaitForSSHStart -VMName $VMName -HvServer $HvServer `
+    -VmPort $VmPort -User $VMUserName -Password $VMPassword -StepTimeout 360
+    RemoteCopy -download -downloadFrom $newIP -files "/home/${VMUserName}/state.txt" `
+        -downloadTo $LogDir -port $VMPort -username $VMUserName -password $VMPassword
+    RemoteCopy -download -downloadFrom $newIP -files "/home/${VMUserName}/BVT-NET-IFUP-IFDOWN.log" `
+        -downloadTo $LogDir -port $VMPort -username $VMUserName -password $VMPassword
+    $contents = Get-Content -Path $stateFile
+    if (($contents -eq "TestAborted") -or ($contents -eq "TestFailed")) {
+        LogErr "Error: Running $remoteScript script failed on VM!"
```
Comment (adriansuhov) [inline Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1:41]: Remove "Error: " from here and all places and leave just the message. LogErr already outputs Error: .. msg..

File: Testscripts/Windows/FCOPY-NEGATIVE.ps1 [hunk 4]
```diff
@@ -83,15 +83,37 @@ function Main {
...
+        return "ABORTED"
     }
+    if (-not $gsi.Enabled) {
+        LogWarn "The Guest services are not enabled for VM '${VMname}'"
+        if ((Get-VM -ComputerName $HvServer -Name $VMname).State -ne "Off") {
+            Stop-VM -ComputerName $HvServer -Name $VMname -Force -Confirm:$false
+        }
+        # Waiting until the VM is off
+        while ((Get-VM -ComputerName $HvServer -Name $VMname).State -ne "Off") {
+            LogMsg "Turning off VM:'${VMname}'"
```
Comment (adriansuhov) [inline Testscripts/Windows/FCOPY-NEGATIVE.ps1:101]: I don't think '${VMname}' is necessary here, ${VMname} should do it.  Also take attention to variable naming/casing. It is declared as $VMName but is used as $VMname. Check all the occurences of this

File: Testscripts/Windows/FCOPY-NEGATIVE.ps1 [hunk 5]
```diff
@@ -83,15 +83,37 @@ function Main {
...
+        LogMsg "Enabling  Guest services on VM:'${VMname}'"
+        Enable-VMIntegrationService -Name "Guest Service Interface" -VMname $VMname -ComputerName $HvServer
+        LogMsg "Starting VM:'${VMname}'"
+        Start-VM -Name $VMname -ComputerName $HvServer
+        # Waiting for the VM to run again and respond
+        do {
+            Start-Sleep -Seconds 5
+        } until (Test-NetConnection $Ipv4 -Port $VMPort -WarningAction SilentlyContinue | Where-Object { $_.TcpTestSucceeded } )
+    }
+
```
Comment (adriansuhov) [inline Testscripts/Windows/FCOPY-NEGATIVE.ps1]: double new line

**Review by adriansuhov — CHANGES_REQUESTED** [review]

File: Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1 (hunk 1 above)
Comment (asofron) [reply Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1]: Fixed

File: Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1 (hunk 3 above)
Comment (asofron) [reply Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1:41]: Fixed

File: Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1 (hunk 2 above)
Comment (asofron) [reply Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1]: Fixed

File: Testscripts/Windows/FCOPY-NEGATIVE.ps1 (hunk 4 above)
Comment (asofron) [reply Testscripts/Windows/FCOPY-NEGATIVE.ps1:101]: Fixed

File: Testscripts/Windows/FCOPY-NEGATIVE.ps1 (hunk 5 above)
Comment (asofron) [reply Testscripts/Windows/FCOPY-NEGATIVE.ps1]: Fixed

File: Testscripts/Windows/FCOPY-NEGATIVE.ps1 [hunk 6]
```diff
@@ -83,15 +83,35 @@ function Main {
...
+    if (-not $gsi) {
+        LogErr "Unable to retrieve Integration Service status from VM '${VMname}'"
+        return "ABORTED"
+    }
+    if (-not $gsi.Enabled) {
+        LogWarn "The Guest services are not enabled for VM '${VMname}'"
+        if ((Get-VM -ComputerName $HvServer -Name $VMName).State -ne "Off") {
+            Stop-VM -ComputerName $HvServer -Name $VMName -Force -Confirm:$false
+        }
+        # Waiting until the VM is off
```
Comment (chvalean) [inline Testscripts/Windows/FCOPY-NEGATIVE.ps1:99]: if you use forced stop then the while loop below is no longer required...

File: Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1 [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        $VMUserName,
+        $VMPassword,
+        $RootDir
+    )
+
+    $remoteScript = "BVT-NET-IFUP-IFDOWN.sh"
+
+    Set-Location $RootDir
+    #
+    # Run the guest VM side script to verify floppy disk operations
```
Comment (chvalean) [inline Testscripts/Windows/BVT-NET-IFUP-IFDOWN.ps1]: incorrect message

File: Testscripts/Linux/timesync_config.sh [hunk 8]
```diff
@@ -36,6 +36,8 @@ case $DISTRO in
         fi
     ;;
     ubuntu* | debian*)
+        apt-get update
+        #Update required before install
```
Comment (chvalean) [inline Testscripts/Linux/timesync_config.sh:39]: comment before the command

**Pull request closed without merging**